### PR TITLE
feat(web): render slack-notify events in session transcript (PR 6)

### PR DESCRIPTION
**Pull request by ColeMurray**
## Summary

PR 6 / 6 of the agent Slack notification feature ([spec](docs/agent-slack-notification-tool.md), [plan](docs/agent-slack-notification-tool-implementation-plan.md)).

Polishes the session transcript so `slack-notify` tool calls render as a first-class event instead of a generic JSON dump.

- **Success**: Slack icon, channel input on the line, expanded view with a "View in Slack" permalink (target=_blank, rel=noopener noreferrer) and subtle notes for `truncated` / `strippedBroadcasts` / `mentionsModified`.
- **Denial**: error icon plus reason-specific copy:
  - `channel_not_found_or_forbidden` → "invite the Open-Inspect bot to the channel" hint
  - `feature_disabled` → "disabled for this repository"
  - `rate_limited` → retry-window copy
  - `feature_unavailable`, `empty_message_after_sanitization`, `slack_api_error`, `invalid_input` → tailored headlines
- **Unparseable output**: falls back to "No details available" rather than crashing.

`ToolCallItem` dispatches early on `event.tool === "slack-notify"` and delegates to the new `SlackNotifyEvent`, so every other tool (Read/Edit/Bash/Apply Patch/etc.) keeps its existing rendering verbatim.

The plan referenced files (`event-list.tsx`, `session/` subdir) that don't exist in this repo — the actual transcript rendering goes through `tool-call-item.tsx` + `formatToolCall`. Wiring follows the existing dispatcher pattern instead.

## Files

- `packages/web/src/components/slack-notify-event.tsx` (new) — collapsed line + expanded body, success / denial / fallback states
- `packages/web/src/components/slack-notify-event.test.tsx` (new) — 7 tests
- `packages/web/src/components/tool-call-item.tsx` — early-return delegation when `event.tool === "slack-notify"`
- `packages/web/src/components/ui/icons.tsx` — `SlackIcon`

## Test plan

- [x] `npm test -w @open-inspect/web` — **228 passed** (7 new in `slack-notify-event.test.tsx`)
- [x] `npm run typecheck` — clean across all packages
- [x] `npm run lint` — clean
- [ ] Manual browser smoke (replay a session with a successful post and a denial — both render correctly) — to be done at feature-merge time per plan §"E2E smoke"

## Verifies

- Acceptance gate per plan PR 6: success permalink, denial copy, truncation marker visible, no permalink on denial.
- No regression for any other tool (`tool-call-item` only takes the new path for `slack-notify`).

🤖 Generated with [Claude Code](https://claude.com/claude-code)

## Summary by CodeRabbit

* **New Features**
  * Added Slack notification event display showing success, permission denied, disabled, and rate-limited states, with contextual messages, optional timestamp, and a Slack icon.
  * Integrated specialized rendering for Slack notification events in the tool-call list and added expand/collapse with toggle callback.

* **Tests**
  * Added comprehensive tests covering success and error scenarios, external-link safety, flag notes, edge cases, and expansion behavior.

[Review Change Stack](https://app.coderabbit.ai/change-stack/ColeMurray/background-agents/pull/611)

### CODE REVIEW

File: packages/web/src/components/slack-notify-event.tsx [hunk 1]
```diff
@@ -0,0 +1,205 @@
...
+    <div className="space-y-2">
+      <div>
+        <div className="text-muted-foreground mb-1 font-medium">Channel</div>
+        <div className="text-foreground">{success.channelInput}</div>
+      </div>
+      {success.permalink ? (
+        <div>
+          <div className="text-muted-foreground mb-1 font-medium">Slack message</div>
+          <a
+            href={success.permalink}
```
[the remark this answers is not in the document]
Comment (ColeMurray) [reply packages/web/src/components/slack-notify-event.tsx]: Done — addressed in 65855c2a. Wrapped `success.permalink` with `getSafeExternalUrl()` so any non-https / unparseable scheme falls through the same null path that already hides the link, matching `action-bar.tsx`, `metadata-section.tsx`, `tunnel-urls-section.tsx`, and `code-server-section.tsx`. Added a test that a `javascript:` permalink results in no rendered `<a>`.

File: packages/web/src/components/tool-call-item.tsx [hunk 2]
```diff
@@ -49,6 +50,10 @@ function ToolIcon({ name }: { name: string | null }) {
 }
 
 export function ToolCallItem({ event, isExpanded, onToggle, showTime = true }: ToolCallItemProps) {
+  if (event.tool === "slack-notify") {
```
[the remark this answers is not in the document]
Comment (ColeMurray) [reply packages/web/src/components/tool-call-item.tsx:53]: Done — addressed in 65855c2a. Made `SlackNotifyEvent` fully controlled via `isExpanded` / `onToggle` props and dropped the local `useState`, so the row now follows the same contract `ToolCallGroup`'s `expandedItems` set drives for every other tool. Updated the dispatch in `tool-call-item.tsx` and the component tests.

File: packages/web/src/components/tool-call-item.tsx [hunk 3]
```diff
@@ -49,6 +50,10 @@ function ToolIcon({ name }: { name: string | null }) {
...
+    return <SlackNotifyEvent event={event} showTime={showTime} />;
+  }
```
[the remark this answers is not in the document]
Comment (ColeMurray) [reply packages/web/src/components/tool-call-item.tsx:62]: Done — addressed in 65855c2a. `SlackNotifyEvent` is now fully controlled (`isExpanded` + `onToggle` required props, no local state). `ToolCallItem` threads them through, matching the rest of the dispatcher. Tests updated to drive the controlled toggle.

**Comment by ColeMurray** [comment]
### Review feedback addressed in 65855c2a

- **[Security] permalink validation** — `success.permalink` now goes through `getSafeExternalUrl()` before being rendered as `href`, falling back to no-link for non-https or unparseable URLs. Matches `action-bar.tsx`, `metadata-section.tsx`, `tunnel-urls-section.tsx`, and `code-server-section.tsx`. New test: a `javascript:` permalink produces no rendered `<a>`.
- **[Functionality] controlled expansion** — `SlackNotifyEvent` is now fully controlled via `isExpanded` / `onToggle` props (no local `useState`), so it follows the same contract `ToolCallGroup`'s `expandedItems` set drives for every other tool row. Tests updated to drive the controlled toggle and assert `onToggle` is invoked on click.
- **[Nit] `noreferrer` assertion** — added `expect(link).toHaveAttribute("rel", expect.stringContaining("noreferrer"))` alongside the existing `noopener` check.

Gates after the fix: 230 web tests / 29 files green; typecheck and lint clean.
